deployment/simFullProposal.js

```diff
@@ -0,0 +1,146 @@
+require('@nomiclabs/hardhat-ethers');
+const path = require('path');
+const fs = require('fs');
+
+let kyberGov;
+
+let forkParams = {};
+let admin;
+let daoOperator;
+let kyberGovAddress;
+let executorAddress;
+let voterAddresses = [
+  '0xf977814e90da44bfa03b6295a0616a897441acec',
+  '0x3Be35d6C5FFeAe62CB3f6CB8a23653b6501A060d',
+  '0x0bfEc35a1A3550Deed3F6fC76Dde7FC412729a91',
+  '0x06890D4c65A4cB75be73D7CCb4a8ee7962819E81',
+];
+
+let ONE;
+let oneEth;
+
+let txData;
+let startTimestamp;
+let endTimestamp;
+let proposalId;
+
+task('simFullProposal', 'simulate proposal creation, voting and execution')
+  .addParam('f', 'JSON file for binary proposal data info')
+  .setAction(async (taskArgs) => {
+    const jsonPath = path.join(__dirname, taskArgs.f);
+    const jsonInput = JSON.parse(fs.readFileSync(jsonPath, 'utf8'));
+    parseInput(jsonInput);
+
+    // attempt mainnet forking
+    try {
```

Comment:
if an error is throw, does this task stop and error print to the screen?
If yes, why we need to try-catch here